#!/usr/bin/env python
PACKAGE = "toposens_driver"
NODE_NAME = "toposens_driver_node"
PARAMS_NAME = "TsDriver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# only core commands are sent on init
gen.add("sig_strength", int_t, 1, "Number of waves emitted in every transmission cycle", 5, 0, 20)
gen.add("filter_size", int_t, 2, "Kernel size applied on ADC signals for edge detection", 20, 1, 100)
gen.add("noise_thresh", int_t, 3, "Minimum amplitude for an echo to be considered valid", 5, 0, 20)
gen.add("snr_boost", int_t, 4, "Post-processing filter for boosting signal-to-noise ratio", 500, 0, 1000)

exit(gen.generate(PACKAGE, NODE_NAME, PARAMS_NAME))


# Make sure this file is given execution privleges
# chmod a+x cfg/Ts*.cfg
